IdentityCore/src/requests/MSIDInteractiveAuthorizationCodeRequest.m

```diff
@@ -160,25 +162,19 @@ - (void)getAuthCodeWithCompletionImpl:(MSIDInteractiveAuthorizationCodeCompletio
         }
         else if ([response isKindOfClass:MSIDWebOpenBrowserResponse.class])
         {
-            NSURL *browserURL = ((MSIDWebOpenBrowserResponse *)response).browserURL;
-
-#if TARGET_OS_IPHONE
-            if (![MSIDAppExtensionUtil isExecutingInAppExtension])
+            NSError *error = nil;
+            MSIDWebResponseBaseOperation *operation = [MSIDWebResponseOperationFactory createOperationForResponse:response error:&error];
+            if (error)
             {
-                MSID_LOG_WITH_CTX_PII(MSIDLogLevelInfo, nil, @"Opening a browser - %@", MSID_PII_LOG_MASKABLE(browserURL));
-                [MSIDAppExtensionUtil sharedApplicationOpenURL:browserURL];
+                returnErrorBlock(error);
+                return;
             }
-            else
+            
+            BOOL isCurrentFlowFinished = [operation doActionWithCorrelationId:self.requestParameters.correlationId error:&error];
+            if (isCurrentFlowFinished && error)
```

Comment:
should we create an unexpected error in the else case? otherwise, the flow will hang.

Reply:
Very good catch here. Updated the code and add a new error just in case it will fall into this part